DHFPROD-2531: Add snackbar saved indicator

wooldridge opened the pull request:
Add snackbar saved indicator for UI changes that lack feedback:

- Ingestion step config
- Mapping step config
- Custom step config

marklogic-builder commented:
Jenkins Build failure 
https://jenkins.marklogic.com/job/5.x/job/dhf-QS-develop-rh7-Commit/2391/testReport
 45 tests run, 4 skipped, 2 failed.

brucean52 commented on web/src/main/ui/app/components/flows-new/edit-flow/custom/ui/custom-ui.component.ts:
```diff
@@ -11,16 +12,10 @@ export class CustomUiComponent {
   onChange() {
     this.updateCustom.emit(this.step);
-  }
+    this.snackBar.open("Change Saved", "", {panelClass: ['snackbar'], duration: 1200});  }
```
the snackbar open code should be in the edit-flow.component.ts where @briantang originally had it. This way the snackbar doesn't appear until we get the response from the server.

marklogic-builder commented:
Jenkins Build failure 
https://jenkins.marklogic.com/job/5.x/job/dhf-QS-develop-rh7-Commit/2404/testReport
 45 tests run, 4 skipped, 32 failed.

brucean52 commented on web/src/main/ui/app/components/flows-new/edit-flow/edit-flow.component.ts:
```diff
@@ -175,6 +192,9 @@ export class EditFlowComponent implements OnInit, OnDestroy {
+      this.snackBar.open("Change Saved", "", {panelClass: ['snackbar'], duration: 1200});
+    }
     this.manageFlowsService.updateStep(this.flow.id, step.id, step).subscribe(resp => {
       this.stepsArray.forEach( step => {
```
snackbar code should be after the resp is received.

wooldridge commented:
@ayuwono the mastering config is a bit different in that the user gets feedback upon saves through the dialog closure and the timestamp button changing. I need to discuss with PM.

marklogic-builder commented:
Jenkins Build failure 
https://jenkins.marklogic.com/job/5.x/job/dhf-QS-develop-rh7-Commit/2413/testReport
 45 tests run, 4 skipped, 2 failed.

ayuwono commented:
@wooldridge, I think there are some conflicts to be resolved in order to merge this

wooldridge commented:
Discussed with @sbayatpur. see: https://project.marklogic.com/jira/browse/DHFPROD-2531

 Added snackbar feedback for Mastering configuration. Now also step configuration has consistent saved feedback. Also updated snackbar labeling to "Step Saved".

marklogic-builder commented:
Jenkins Build failure 
https://jenkins.marklogic.com/job/5.x/job/dhf-QS-develop-rh7-Commit/2427/testReport
 45 tests run, 4 skipped, 32 failed.

wooldridge commented:
Also moved snackbar generation to edit-flow-ui.component since it is a UI feature. It is now called from edit-flow.component via @ViewChild.

wooldridge commented:
@ayuwono you bug you mention is separate from the feature in this PR. Please file a separate bug so we can merge this. Thanks for catching!

ayuwono commented:
> @ayuwono you bug you mention is separate from the feature in this PR. Please file a separate bug so we can merge this. Thanks for catching!

sounds good, will log a separate issue
